src/Spice86/UserControls/AddressAutoCompleteBox.axaml.cs

```diff
@@ -0,0 +1,128 @@
+using Avalonia;
+using Avalonia.Controls;
+
+using Spice86.Core.Emulator.CPU;
+using Spice86.Core.Emulator.Memory;
+using Spice86.Shared.Utils;
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace Spice86.UserControls;
+
+public partial class AddressAutoCompleteBox : UserControl {
+    public AddressAutoCompleteBox() {
+        InitializeComponent();
+        this.AddressTextBox.TextChanged += OnTextChanged;
+    }
+
+    private void OnTextChanged(object? sender, TextChangedEventArgs e) {
+        OnTextChanged(AddressTextBox.Text);
+    }
+
+    private bool TryParseMemoryAddress(string? addressExpression,
+        [NotNullWhen(true)] out uint? address) {
+        if (string.IsNullOrWhiteSpace(addressExpression)) {
+            address = null;
+            return false;
+        }
+
+        try {
+            if (addressExpression.Contains(':')) {
+                string[] split = addressExpression.Split(":");
+                string firstPart = split[0];
+                string secondPart = split[1];
+
+                if(State is not null) {
+                    firstPart = ReplaceRegisterNameWithValue(firstPart, State);
+                    secondPart = ReplaceRegisterNameWithValue(secondPart, State);
+                }
+
+                if (split.Length > 1 &&
+                    ushort.TryParse(firstPart, NumberStyles.HexNumber,
+                    CultureInfo.InvariantCulture, out ushort hexSegment) &&
+                    ushort.TryParse(secondPart, NumberStyles.HexNumber,
+                    CultureInfo.InvariantCulture, out ushort hexOffset)) {
+                    address = Math.Min(A20Gate.EndOfHighMemoryArea, MemoryUtils.
+                        ToPhysicalAddress(hexSegment, hexOffset));
+                    return true;
+                }
+
+                if (split.Length > 1 &&
+                    ushort.TryParse(firstPart, NumberStyles.Integer,
+                    CultureInfo.InvariantCulture, out ushort segment) &&
+                    ushort.TryParse(secondPart, NumberStyles.Integer,
+                    CultureInfo.InvariantCulture, out ushort offset)) {
+                    address = Math.Min(A20Gate.EndOfHighMemoryArea, MemoryUtils.
+                        ToPhysicalAddress(segment, offset));
+                    return true;
```

Comment:
I don't understand.  Any input that passes this surely also passed the previous check, so this code will never do anything?

Reply:
File removed.